GeomIO/SharedIO.F90

```diff
@@ -13,9 +14,128 @@ module mapl3g_SharedIO
    public get_mapl_geom
    public create_time_variable
    public bundle_to_metadata
+   public esmf_to_pfio_type
+   public create_local_start
+   public create_global_count
+   public create_global_start
+   public create_file_shape
 
    contains
 
+   function create_file_shape(grid, field_shape, rc) result(file_shape)
+      integer, allocatable :: file_shape(:)
+      type(ESMF_Grid), intent(in) :: grid
+      integer, intent(in) :: field_shape(:)
+      integer, intent(out), optional :: rc
+
+      integer :: status, sz, ungr, tile_count
+      call ESMF_GridGet(grid, tileCount=tile_count, _RC)
+      sz = size(field_shape)
+      ungr = sz - 2
```

Comment:
Perhaps I'm missing the constraints on the call side, but is this always true?

Reply:
For grids yes, the rank - 2 is the number ungridded dimensions 

If this were for a locstream the constraint would be -1, for example, so this needs generalization

Reply:
Ah.  Yes.  I think this will be clearer if "2" is replaced by a parameter "NUM_GRIDDED_DIMS".  Then clearly 
```
num_ungridded_dims = num_dims - NUM_GRIDDED_DIMS
```
Make it obvious to the next person ...